sklearn/gaussian_process/gpr.py

```diff
@@ -236,6 +236,33 @@ def obj_func(theta, eval_gradient=True):
             # likelihood
             lml_values = list(map(itemgetter(1), optima))
             self.kernel_.theta = optima[np.argmin(lml_values)][0]
+            if np.isclose(self.kernel_.bounds,
```

Comment:
Best to use a local variable to store this mask rather than repeat the code and computation.